src/server/.tests/ws-routes/controllers/Friends.spec.js

```diff
@@ -0,0 +1,497 @@
+					},
+					{
+						$id: 'dummyId2',
+						username: 'dummyFriend2'
+					}
+				]
+			}
+
+			const queryResultMock = {
+				exec: sinon.stub().resolves(QUERY_FEEDBACK_MOCK)
+			}
+
+			suite.userModelMock.findOne.returns(queryResultMock)
+
+			suite.server.on(CLIENT_EVENTS.CONNECTION, connection => {
+				suite.emitSpy = sinon.spy(connection, 'emit')
+				connection.on(CLIENT_EVENTS.FRIENDS_LIST, data => {
+					return suite.friendsInstance.friendsList(data, connection)
+						.then(() => {
+
+							//then
+							const expectedPayload = QUERY_FEEDBACK_MOCK.friends
+							sinon.assert.calledWith(suite.emitSpy, CLIENT_EVENTS.FRIENDS_LIST, expectedPayload)
+							done()
+						})
+				})
+			})
+
+			//when
+			suite.client = socketClient.connect(SERVER_URL, SOCKET_OPTIONS)
+			suite.client.emit(CLIENT_EVENTS.FRIENDS_LIST)
+		})
+	})
+
+	describe('#Notifications', () => {
+		beforeEach(() => {
+			suite.userModelMock = {
+				findOneAndUpdate: sinon.stub()
+			}
+
+			suite.friendsInstance = new FriendsProvider({
+				UserModel: suite.userModelMock
+			})
+		})
+
+		describe('#RemoveNotifications', () => {
+
+			it('should call emit with remove notifications event type when notifications was removed', done => {
+				//given
+				const DUMMY_REQUEST = []
+				const queryResultMock = {
+					exec: sinon.stub().resolves()
+				}
+				suite.userModelMock.findOneAndUpdate.returns(queryResultMock)
+	
+				suite.server.on(CLIENT_EVENTS.CONNECTION, socket => {
+					socket.on(CLIENT_EVENTS.REMOVE_NOTIFICATIONS, data => {
+						suite.emitSpy = sinon.spy(socket, 'emit')
+						suite.friendsInstance.removeNotifications(data, socket)
+							.then(() => {
+								//then
+								sinon.assert.calledWith(suite.emitSpy.firstCall, CLIENT_EVENTS.REMOVE_NOTIFICATIONS)
+								done()
+							})
+					})
+				})
+	
+				//when
+				suite.client = socketClient.connect(SERVER_URL, SOCKET_OPTIONS)
+				suite.client.emit(CLIENT_EVENTS.REMOVE_NOTIFICATIONS, DUMMY_REQUEST)
+			})
+		
+	
+			it('should remove notification records from db by id ', done => {
+				//given
+				const DUMMY_ID_1 = 'dummyId1'
+				const DUMMY_ID_2 = 'dummyId2'
+				const REQUEST_MOCK = {
+					notificationIds : [
+						{
+							_id: DUMMY_ID_1
+						},
+						{
+							_id: DUMMY_ID_2
+						}
+					]
+				}
+	
+				const queryResultMock = {
+					exec: sinon.stub().resolves()
+				}
+				suite.userModelMock.findOneAndUpdate.returns(queryResultMock)
+	
+	
+				suite.server.on(CLIENT_EVENTS.CONNECTION, socket => {
+					socket.on(CLIENT_EVENTS.REMOVE_NOTIFICATIONS, data => {
+						suite.friendsInstance.removeNotifications(data, socket)
+						then()
+					})
+				})
+	
+				//when
+				suite.client = socketClient.connect(SERVER_URL, SOCKET_OPTIONS)
+				suite.client.emit(CLIENT_EVENTS.REMOVE_NOTIFICATIONS, REQUEST_MOCK)
+			
+				//then
+				function then() {
+					const expectedSearchingCriteria = {username: suite.DUMMY_USERNAME}
+					const expectedQuery = {
+						$pull: {
```

Comment:
Could we make test that checks if records were really removed?
Otherwise change name  of test case, so others will know that it's just testing if query has been sent to Mongo.